examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/main.c

```diff
@@ -78,180 +80,298 @@
+        return -1;
+
+    if ( 0 > build_xively_topic( MAGNETOMETER_CHANNEL_NAME, xi_account_id, xi_device_id,
+                                 mqtt_topics.magnetometer, XI_TOPIC_MAX_LEN ) )
+        return -1;
+
+    if ( 0 > build_xively_topic( BAROMETER_CHANNEL_NAME, xi_account_id, xi_device_id,
+                                 mqtt_topics.barometer, XI_TOPIC_MAX_LEN ) )
+        return -1;
+
+    if ( 0 > build_xively_topic( HUMIDITY_CHANNEL_NAME, xi_account_id, xi_device_id,
+                                 mqtt_topics.hygrometer, XI_TOPIC_MAX_LEN ) )
+        return -1;
+
+    if ( 0 > build_xively_topic( TEMPERATURE_CHANNEL_NAME, xi_account_id, xi_device_id,
+                                 mqtt_topics.temperature, XI_TOPIC_MAX_LEN ) )
+        return -1;
+
+    if ( 0 > build_xively_topic( BUTTON_CHANNEL_NAME, xi_account_id, xi_device_id,
+                                 mqtt_topics.button, XI_TOPIC_MAX_LEN ) )
+        return -1;
+
+    if ( 0 > build_xively_topic( LED_CHANNEL_NAME, xi_account_id, xi_device_id,
+                                 mqtt_topics.led, XI_TOPIC_MAX_LEN ) )
+        return -1;
+
+    return 0;
+}
+
+xi_state_t pub_accelerometer( void )
 {
-    char out_msg[IO_AXES_JSON_BUFFER_MAX_SIZE] = "";
+    char out_msg[IO_AXES_JSON_BUFFER_MAX_SIZE];
     SensorAxes_t sensor_input;
 
-    if ( io_read_accelero( &sensor_input ) < 0 )
+    if ( 0 > io_read_accelero( &sensor_input ) )
     {
-        printf( "\r\n\t[ERROR] trying to read accelerometer input" );
         return -1;
     }
-    if ( io_axes_to_json( sensor_input, out_msg, IO_AXES_JSON_BUFFER_MAX_SIZE ) < 0 )
+    if ( 0 > io_axes_to_json( sensor_input, out_msg, IO_AXES_JSON_BUFFER_MAX_SIZE ) )
     {
-        printf( "\r\n\t[ERROR] trying to create JSON string from sensor input" );
         return -1;
     }
 
-    return xi_publish( gXivelyContextHandle, topic_name, out_msg,
+    return xi_publish( gXivelyContextHandle, mqtt_topics.accelerometer, out_msg,
                        XI_MQTT_QOS_AT_MOST_ONCE, XI_MQTT_RETAIN_FALSE, NULL, NULL );
 }
 
-xi_state_t pub_magnetometer( const char* topic_name )
+xi_state_t pub_magnetometer( void )
 {
-    char out_msg[IO_AXES_JSON_BUFFER_MAX_SIZE] = "";
+    char out_msg[IO_AXES_JSON_BUFFER_MAX_SIZE];
     SensorAxes_t sensor_input;
 
-    if ( io_read_magneto( &sensor_input ) < 0 )
+    if ( 0 > io_read_magneto( &sensor_input ) )
     {
-        printf( "\r\n\t[ERROR] trying to read magnetometer input" );
         return -1;
     }
-    if ( io_axes_to_json( sensor_input, out_msg, IO_AXES_JSON_BUFFER_MAX_SIZE ) < 0 )
+    if ( 0 > io_axes_to_json( sensor_input, out_msg, IO_AXES_JSON_BUFFER_MAX_SIZE ) )
     {
-        printf( "\r\n\t[ERROR] trying to create JSON string from sensor input" );
         return -1;
     }
-    return xi_publish( gXivelyContextHandle, topic_name, out_msg,
+    return xi_publish( gXivelyContextHandle, mqtt_topics.magnetometer, out_msg,

```

Comment:
here I don't see the benefit storing all the topic strings in a struct variable instead of using the previous macro: XI_TOPIC_NAME_MANGLE( "Magnetometer" ). Is there benefit?